backend/dataall/modules/redshift_datasets/db/redshift_dataset_repositories.py

```diff
@@ -0,0 +1,140 @@
+import logging
+
+from sqlalchemy import or_, and_
+from dataall.core.activity.db.activity_models import Activity
+from dataall.core.environment.db.environment_models import Environment
+from dataall.core.organizations.db.organization_repositories import OrganizationRepository
+from dataall.base.db import paginate
+from dataall.base.db.exceptions import ObjectNotFound
+from dataall.modules.datasets_base.services.datasets_enums import ConfidentialityClassification, Language
+from dataall.core.environment.services.environment_resource_manager import EnvironmentResource
+from dataall.modules.redshift_datasets.db.redshift_models import RedshiftDataset, RedshiftTable
+
+logger = logging.getLogger(__name__)
+
+
+class RedshiftDatasetEnvironmentResource(EnvironmentResource):
+    """Actions performed on any environment resource on environment operations"""
+
+    @staticmethod
+    def count_resources(session, environment, group_uri) -> int:
+        return RedshiftDatasetRepository.count_environment_group_datasets(session, environment, group_uri)
+
+
+class RedshiftDatasetRepository:
+    """DAO layer for Redshift Datasets"""
+
+    _DEFAULT_PAGE = 1
+    _DEFAULT_PAGE_SIZE = 10
+
+    @classmethod
+    def create_redshift_dataset(cls, session, username, env: Environment, data: dict) -> RedshiftDataset:
+        organization = OrganizationRepository.get_organization_by_uri(session, env.organizationUri)
+        dataset = RedshiftDataset(
+            label=data.get('label'),
+            owner=username,
+            description=data.get('description', 'No description provided'),
+            tags=data.get('tags', []),
+            AwsAccountId=env.AwsAccountId,
+            SamlAdminGroupName=data['SamlAdminGroupName'],
+            region=env.region,
+            environmentUri=env.environmentUri,
+            organizationUri=env.organizationUri,
+            language=data.get('language', Language.English.value),
+            confidentiality=data.get('confidentiality', ConfidentialityClassification.Unclassified.value),
+            topics=data.get('topics', []),
+            businessOwnerEmail=data.get('businessOwnerEmail', ''),
+            businessOwnerDelegationEmails=data.get('businessOwnerDelegationEmails', []),
+            stewards=data.get('stewards') if data.get('stewards') else data['SamlAdminGroupName'],
+            autoApprovalEnabled=data.get('autoApprovalEnabled', False),
+            connectionUri=data.get('connectionUri'),
+            schema=data.get('schema'),
+        )
+        session.add(dataset)
+        session.commit()
```

Comment:
shouldn't we commit after the last add to make sure both succeed or get reverted?

Reply:
I implemented the change but the tests started failing. The issue is that if the object dataset is not commited the unique identifier is not created and then the activity creation gets a null value in `targetUri=dataset.datasetUri`